api/integrations/slack/authentication.py

```diff
@@ -7,21 +7,28 @@
 
 from users.models import FFAdminUser
 
+logger = logging.getLogger(__name__)
+
 
 class OauthInitAuthentication(BaseAuthentication):
     """
     Custom authentication class to use signed user_id present in query params
     """
 
     def authenticate(self, request):
+        logger.debug("OauthInitAuthentication authenticate called")
         signer = TimestampSigner()
-        with suppress(BadSignature, ObjectDoesNotExist):
+        try:
             signature = request.GET.get("signature")
             if not signature:
                 raise AuthenticationFailed(
                     "Authentication credentials were not provided."
                 )
             user_id = signer.unsign(signature, max_age=30)
             user = FFAdminUser.objects.get(id=user_id)
+
+            logger.debug(f"OauthInitAuthentication normal return with user {user}")
             return user, None
-        raise AuthenticationFailed("No such user")
+        except (BadSignature, ObjectDoesNotExist):
+            logger.debug("OauthInitAuthentication raising an exception")
+            raise AuthenticationFailed("No such user")
```

Comment:
I think we should keep this out of the except? We should always raise AuthenticationFailed, right? So like: 

```python
try:
    ...
except (BadSignature, ObjectDoesNotExist):
   logger.debug(...)

raise AuthenticationFailed("No such user")
```

Maybe we should even specify which exception was raised? So like: 

```python
except (BadSignature, ObjectDoesNotExist) as e:
    logger.debug("OauthInitAuthentication raising exception: %s", e)
```